x/posts/internal/keeper/keeper.go

```diff
@@ -195,6 +195,72 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 	return filteredPosts
 }
 
+func (k Keeper) getAnswersStoreKey(postID types.PostID) []byte {
+	return []byte(types.PollAnswersStorePrefix + postID.String())
+}
+
+// Save the userAnswersDetails associated with the given postID inside the current context
+// It assumes that the post exists and has a Poll inside it.
+// If userAnswersDetails are already present, the old ones will be overridden.
+func (k Keeper) SavePollUserAnswers(ctx sdk.Context, postID types.PostID, userPollAnswers types.AnswersDetails) {
+	store := ctx.KVStore(k.StoreKey)
+
+	usersAnswersDetails := k.GetPostPollAnswersDetails(ctx, postID)
+
+	if usersAnswersDetails, appended := usersAnswersDetails.AppendIfMissingOrIfUsersEquals(userPollAnswers); appended {
+		store.Set(k.getAnswersStoreKey(postID), k.Cdc.MustMarshalBinaryBare(&usersAnswersDetails))
+	}
+
+}
+
+// GetAllPostPollsAnswers returns the list of all the post polls answers associated with the given postID that are stored into the current state.
+func (k Keeper) GetPostPollAnswersDetails(ctx sdk.Context, postID types.PostID) types.UsersAnswersDetails {
+	store := ctx.KVStore(k.StoreKey)
+
+	var usersAnswersDetails types.UsersAnswersDetails
+	answersBz := store.Get(k.getAnswersStoreKey(postID))
+
+	k.Cdc.MustUnmarshalBinaryBare(answersBz, &usersAnswersDetails)
+
+	return usersAnswersDetails
+}
+
+// GetReactions allows to returns the list of reactions that have been stored inside the given context
+func (k Keeper) GetAnswersDetailsMap(ctx sdk.Context) map[types.PostID]types.UsersAnswersDetails {
+	store := ctx.KVStore(k.StoreKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte(types.PollAnswersStorePrefix))
+
+	usersAnswersData := map[types.PostID]types.UsersAnswersDetails{}
+	for ; iterator.Valid(); iterator.Next() {
+		var userAnswers types.UsersAnswersDetails
+		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &userAnswers)
+		postID, _ := types.ParsePostID(strings.TrimPrefix(string(iterator.Key()), types.PollAnswersStorePrefix))
+		usersAnswersData[postID] = userAnswers
+	}
+
+	return usersAnswersData
+}
+
+// GetPostPollAnswersByUser retrieves post poll answers associated to the given ID and filtered by user
+func (k Keeper) GetPostPollAnswersByUser(ctx sdk.Context, postID types.PostID, user sdk.AccAddress) []uint {
```

Comment:
I don't think we need this anymore

Reply:
I wrote this method to keep the code more readable inside `handler.go` should I remove it?

Reply:
I've seen its usage now. No, we can skip it. I'll close this.